org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/archetype/package-info.java

```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
```

Comment:
We usually don't use package-info.json. Is it something really needed and useful here? If not, I suggest we remove it and maybe add it later, when there is a compelling reason to add them everywhere in m2e.

Reply:
This is `package-info.java` not `.json`. It is necessary to version on package level as recommended by OSGi. Otherwise you cannot do proper semantic versioning as outlined in https://docs.osgi.org/whitepaper/semantic-versioning/040-semantic-versions.html.

Reply:
> It is necessary to version on package level as recommended by OSGi. Otherwise you cannot do proper semantic versioning as outlined in https://docs.osgi.org/whitepaper/semantic-versioning/040-semantic-versions.html.

package version has to be declared in MANIFEST.MF, not with a package-info.java. There might be some tools to build a good MANIFEST with this file, but the MANIFEST-first approach used in m2e doesn't allow that and versions need to be set directly in the MANIFEST.MF Export-Package.

Reply:
Is the ConsumerType/ProviderType annotations evaluated by PDE for automatically generating the correct import-version ranges? Are there any plans for Tycho/PDE to evaluate the official annotations?
I will update the PR accordingly to export this package in version 1.0.0 manually in MANIFEST.MF.

Reply:
> package version has to be declared in MANIFEST.MF, not with a package-info.java. There might be some tools to build a good MANIFEST with this file, but the MANIFEST-first approach used in m2e doesn't allow that and versions need to be set directly in the MANIFEST.MF Export-Package.

As noted above this is not true anymore PDE already handles these annotations and updates the manifest automatically.
Beside that, it allows other tools to find out what is the purpose of an interface if it is used as a dependency so using these annotation can (and should be) done to move forward towards best OSGi practice.

Reply:
Good to know that modification can be automated, but still the MANIFEST.MF does need to include the version as part of the commit as PDE doesn't run those "fixes" in regular Tycho build, does it?

Reply:
Thanks, I updated it in https://github.com/eclipse-m2e/m2e-core/pull/922/commits/4685cbd46790097615112e789598bf0315e1d3a8 to manually set the package export version in the manifest and removed the package-info.java for now.

Reply:
> Good to know that modification can be automated, but still the MANIFEST.MF does need to include the version as part of the commit as PDE doesn't run those "fixes" in regular Tycho build, does it?

Yes, that currently only a very rough first step to empower further enhancements.